City-of-Helsinki/drupal-hdbt
# UHF-3635: Better icons, less mess

# [UHF-3635](https://helsinkisolutionoffice.atlassian.net/browse/UHF-3635)

## What was changed

* Updated icon mixins (pseudo-icon)
* Moved icons to two folder structures 
  * icons
  	* All icons under here get compiled to the svg sprite and filenames need to be unique
  * icons_unused
    * All icons that are available from hds or elsewhere, but not in use are stored here.
    * If you need to include a new file from here, just move it to the correct location in `icons` folder and run `npm run build`
* All icons in css are now referenced through `icon_url()` function
* All icons are served through a single svg sprite file
  * The file has a 5 character hash in its name to cache bust old versions
  * All icons are stacked on top of each other and can be viewed [using their target hash](https://raw.githubusercontent.com/City-of-Helsinki/drupal-hdbt/37493dc8162a02bf2971c4c5060a5ede8dfd0c7f/dist/icons/sprite-5f56a.svg#alert-circle)
* Old duplicate and deprecated icon embeds, from html and css were removed
* Icons are now served using css custom properties to keep filesize smaller and more flexible
  * Caveat: Chrome with developer tools open and cache disabled flashes icons on hover and transitions (it redownloads icons)
* All icons were updated to the latest from HDS
* The old `--icon-proportions` variable was converted to more semantic and easy `--icon-size` variable
* Until we can fix the hdbt buttons from user created content, we need to have --hdbt-icon and --hel-icon.
  * When the buttons are fixed from ckeditor content, we can convert all icons to use --hdbt-icon format
* By default icons get their size from line-height of the text next to them. This reduces the need for setting icon size manually.
* the `src` folder no longer gets compiled content in it. Only dist folder is automatically adjusted.
* Admin selectable icon list was culled and not needed icons were removed in collaboration with design team.
* The library for creating svg sprites was switched to a more flexible, patched svg-sprite version (that will get a PR at a later time with the patch)


Other changes:
* Language switcher has now a small delay before it pops up in view when scrolling back. This reduces page jumping
* React and Share email symbol was fixed (it now has a similar round box like other social media icons)
* Helsinki logo dimensions were fixed for all translations
* Animations of icons were unified using a common variable
* hds styles are not embedded from node_modules, they're copied to our codebase to make transition to our own code easier

See: 
* https://github.com/City-of-Helsinki/drupal-hdbt-admin/pull/102
* https://github.com/City-of-Helsinki/drupal-helfi-platform/pull/56
* https://github.com/City-of-Helsinki/drupal-helfi-sote/pull/275
* https://github.com/City-of-Helsinki/drupal-helfi-kasvatus-koulutus/pull/44

## How to test / install
- Make sure your instance is up and running on latest dev branch
- Update the modules/themes
   - `composer require drupal/hdbt:dev-UHF-3635_better_icons && composer require drupal/hdbt_admin && composer require drupal/hdbt_admin:dev-UHF-3635_better_icons_builder`
- Run `make drush-updb drush-cr`
- Log in to the site and check all icons everywhere (sorry)
